caikit/runtime/http_server/http_server.py

```diff
@@ -532,6 +551,50 @@ def _health_check() -> str:
         log.debug4("Server healthy")
         return "OK"
 
+    @contextmanager
+    def _tls_files(self) -> _TlsFiles:
+        """This contextmanager ensures that the tls config values are files on
+        disk since SslContext requires files
+
+        Returns:
+            tls_files (_TlsFiles): The set of configured TLS files
+        """
+        tls_cfg = self.tls_config or {}
+        tls_opts = {
+            "server_key": tls_cfg.get("server", {}).get("key", ""),
+            "server_cert": tls_cfg.get("server", {}).get("cert", ""),
+            "client_cert": tls_cfg.get("client", {}).get("cert", ""),
+        }
+        non_file_opts = {
+            key: val for key, val in tls_opts.items() if val and not os.path.isfile(val)
+        }
+        if not non_file_opts:
+            log.debug3("No TLS inline files configured")
+            yield _TlsFiles(**tls_opts)
+            return
+        # If any of the values are set and are not pointing to files, save them
+        # to a temporary directory
+        try:
+            with tempfile.TemporaryDirectory() as tls_dir:
+                updated_files = {
+                    key: val
+                    for key, val in tls_opts.items()
+                    if key not in non_file_opts and val
+                }
+                for fname, content in non_file_opts.items():
+                    temp_fname = os.path.join(tls_dir, fname)
+                    updated_files[fname] = temp_fname
+                    with open(temp_fname, "w", encoding="utf-8") as handle:
+                        handle.write(content)
+                yield _TlsFiles(**updated_files)
+        except OSError as err:
+            log.error(
+                "<RUN80977064E>",
+                "Cannot create temporary TLS files. Either pass config as file paths or run with write permissions.",
```

Comment:
oof :(
Right, it's probably not a safe assumption that a container on openshift will have write access to `/tmp`...

There's not any chance that uvicorn kwargs like `tls_kwargs["ssl_keyfile"]` can be passed as open io streams, is there?

Reply:
Right, this is not great for exactly that reason (thus my error handling). The code in `uvicorn` is here: https://github.com/encode/uvicorn/blob/master/uvicorn/config.py#L121

It's all just raw `ssl` library, so we can poke around there, but I'm not optimistic.

Reply:
Yeah, just tried to pass it an `io.StringIO` and it barfs all the same:
```
TypeError: certfile should be a valid filesystem path
```

Reply:
Found it! https://github.com/python/cpython/blob/main/Modules/_ssl.c#L3884C61-L3884C61